tensorflow/python/tfe_wrapper.cc

```diff
@@ -282,6 +282,21 @@ static py::object TF_ListPhysicalDevices() {
   return tensorflow::PyoOrThrow(result);
 }
 
+static py::object TF_ListPluggablePhysicalDevices() {
+  std::vector<string> devices;
+  tensorflow::Status s =
+      tensorflow::DeviceFactory::ListPluggablePhysicalDevices(&devices);
+  MaybeRaiseRegisteredFromStatus(s);
+  PyObject* result = PyList_New(devices.size());
```

Comment:
Please use a smart pointer (tensorflow::Safe_PyObjectPtr) and just release it when returning.

Code like this often develops early returns over time which can be memory leaks.

Reply:
Done, please have a review. Thanks